pyvista/utilities/__init__.py

```diff
@@ -1,77 +1,39 @@
-"""Utilities routines."""
-from .algorithms import algorithm_to_mesh_handler, set_algorithm_input
-from .common import perlin_noise, sample_function
-from .errors import (
-    GPUInfo,
-    Observer,
-    Report,
-    assert_empty_kwargs,
-    get_gpu_info,
-    send_errors_to_logging,
-    set_error_output_file,
-    check_valid_vector,
-    VtkErrorCatcher,
-)
-from .features import *
-from .fileio import *
-from .geometric_objects import *
-from .helpers import *
-from .parametric_objects import *
-from .sphinx_gallery import Scraper, _get_sg_image_scraper
-from .regression import compare_images
-from . import transformations
-from .xvfb import start_xvfb
-from .reader import (
-    get_reader,
-    AVSucdReader,
-    BaseReader,
-    BinaryMarchingCubesReader,
-    BMPReader,
-    BYUReader,
-    CGNSReader,
-    DEMReader,
-    EnSightReader,
-    FacetReader,
-    FluentReader,
-    GIFReader,
-    GLTFReader,
-    HDFReader,
-    HDRReader,
-    JPEGReader,
-    MetaImageReader,
-    MFIXReader,
-    MultiBlockPlot3DReader,
-    NIFTIReader,
-    NRRDReader,
-    OBJReader,
-    OpenFOAMReader,
-    POpenFOAMReader,
-    Plot3DFunctionEnum,
-    Plot3DMetaReader,
-    PLYReader,
-    PNGReader,
-    PNMReader,
-    PointCellDataSelection,
-    PTSReader,
-    PVDDataSet,
-    PVDReader,
-    SegYReader,
-    SLCReader,
-    STLReader,
-    TecplotReader,
-    TIFFReader,
-    TimeReader,
-    VTKDataSetReader,
-    VTKPDataSetReader,
-    XMLImageDataReader,
-    XMLPImageDataReader,
-    XMLRectilinearGridReader,
-    XMLPRectilinearGridReader,
-    XMLUnstructuredGridReader,
-    XMLPUnstructuredGridReader,
-    XMLPolyDataReader,
-    XMLStructuredGridReader,
-    XMLMultiBlockDataReader,
-    DICOMReader,
-    XdmfReader,
-)
+"""Deprecated utilities module."""
+import importlib
+import inspect
+import warnings
+
+from pyvista.core.errors import DeprecationError, PyVistaDeprecationWarning
+
+
+def __dir__():
+    raise DeprecationError('The `pyvista.utilities` module has been deprecated.')
+
+
+def __getattr__(name):
+    # TODO: for some reason this runs twice
+    utils = importlib.import_module('pyvista.core.utilities')
+    try:
+        value = inspect.getattr_static(utils, name)
+        import_path = f'from pyvista.core.utilities import {name}'
+    except AttributeError:
+        utils = importlib.import_module('pyvista.plotting.utilities')
+        try:
+            value = inspect.getattr_static(utils, name)
+            import_path = f'from pyvista.plotting.utilities import {name}'
+        except AttributeError:
+            raise AttributeError(
+                f'Module `pyvista.utilities` has been deprecated and we could not automatically find `{name}` in `pyvista.core.utilities` or `pyvista.plotting.utilities`.'
+            )
```

Comment:
I'd suppress the exception context so that users only see this latter exception.
```suggestion
            ) from None
```

Reply:
Hmm, I think I posted this to a single commit, let me repost properly. Sorry.